src/node.js

```diff
@@ -390,6 +391,78 @@
     }
   };
 
+  startup.processPromises = function() {
+    // this should be kept in sync with PromiseRejectEvent in v8.h
+    var kPromiseRejectWithNoHandler = 0;
+    var kPromiseHandlerAddedAfterReject = 1;
+
+    var timers = NativeModule.require('timers');
+    var unhandledPromises = new WeakMap();
+    // A hack to work around uniterable WeakMaps.
+    var unhandledPromiseKeys = [];
+    var fullTurnQueued = false;
+
+    function queueFullTurn() {
+      if (!fullTurnQueued) {
+        fullTurnQueued = true;
+        timers.setImmediate(function() {
+          process.nextTick(afterFullTurn);
+        });
+      }
+    }
+
+    function afterFullTurn() {
+      fullTurnQueued = false;
+      var keys = unhandledPromiseKeys;
+      unhandledPromiseKeys = [];
+      for (var i = 0; i < keys.length; ++i) {
+        var promise = keys[i];
+        var unhandledRejection = unhandledPromises.get(promise);
+        // A handler was added before a full turn happened.
+        if (unhandledRejection === undefined) continue;
+        unhandledRejection.hasBeenNotified = true;
+        var reason = unhandledRejection.reason;
+        if (!process.emit('unhandledRejection', reason, promise)) {
+          // Nobody is listening.
```

Comment:
Exactly :) Those are a lot harder to make. The current PR does not dictate any default behavior and doesn't break peoples' workflows.

I'd like to emphasize again that I think we _should_ add a default handler and it would be very useful for the very least to log (for the very least if the debug flag is turned on) - I just think that this decision should be deferred until this PR is merged.